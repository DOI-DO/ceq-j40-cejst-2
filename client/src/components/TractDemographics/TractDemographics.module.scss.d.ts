declare namespace TractDemographicsNamespace {
    export interface ITractDemographicsScss {
      tractDemographicsContainer: string;
    }
  }

declare const TractDemographicsScssModule: TractDemographicsNamespace.ITractDemographicsScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: TractDemographicsNamespace.ITractDemographicsScss;
  };

  export = TractDemographicsScssModule;
