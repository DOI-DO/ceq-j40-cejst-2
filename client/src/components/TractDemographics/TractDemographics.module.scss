@use '../../styles/design-system.scss' as *;
@import "../utils.scss";

.demographicsContainer{
    padding: 1.2rem 1rem 0 1.2rem;
    
    .demographicsLabel {
        @include sidePanelLabelStyle;
    }

    [id*='expand-content']{
        padding: .5rem 0;
        
    }

    .demographicItem {
        display: flex;
        justify-content: space-between;
        padding-bottom: .5rem;
    
        span {
            font-size: .8rem;
            margin-top: 0;
        }
    }
    .customDemographicHeading {
        @include u-margin-top(.5);
        font-size: medium;
    }
}
