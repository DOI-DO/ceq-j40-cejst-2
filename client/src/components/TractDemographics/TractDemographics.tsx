import React, {useState} from 'react';

import * as styles from './TractDemographics.module.scss';

// Mock interface
export interface ITractDemographicsProps {
  racial: [string, number][],
  age: [string, number][]
}

// Mock backend data
const demographicsData:ITractDemographicsProps = {
  racial: [
    ['White', 61.4],
    ['Black or African Americon', 10.3],
    ['American Indian and Alaska Native', 10.3],
    ['American Indian and Alaska Native', 0.3],
    ['Asian', 7.2],
    ['Native Hawiian or Pacific Islander', 4.2],
    ['Other', 2.5],
    ['Two or more races', 6.7],
    ['Hispanic or Latino', 10.4],
  ],
  age: [
    ['Children under 10', 1.4],
    ['Ages 10 - 64', 80.3],
    ['Elderly over 65', 7.2],
  ],
};

/*
 * Generate the demographic item based the input
 * // TODO: Update after actual data is created
 *
 */
const demographicItemGen = (demographicData: any[]) => {
  return demographicData.map((el, index) => {
    return (
      <div key={index} className={styles.demographicItem}>
        <span>{ el[0] }</span>
        <span>{`${el[1]}%`}</span>
      </div>
    );
  });
};

interface IJ40AccordionItem {
  id: string,
  title: string,
  children: React.ElementType
}

const J40AccordionItem = ({id, title, children}:IJ40AccordionItem) => {
  const [isExpanded, setIsExpanded] = useState(false);
  return (

    <>
      <h6 className={styles.customDemographicHeading}>
        {title}
        <button
          id={`${id}-expand-control`}
          type="button"
          className="usa-accordion__button usa-banner__button"
          aria-expanded={isExpanded}
          aria-controls={`${id}-expand-content`}
          onClick={() => setIsExpanded(!isExpanded)}
        >
          <span className="usa-banner__button-text">{isExpanded ? `hide` : 'show'}</span>
        </button>
      </h6>

      <div
        id={`${id}-expand-content`}
        className="usa-banner__content usa-accordion__content"
        aria-labelledby={`${id}-expand-control`}
        hidden={!isExpanded}
      >
        {children}
      </div>
    </>
  );
};

const TractDemographics = () => {
  return (
    <div className={styles.demographicsContainer}>
      <div className={styles.demographicsLabel}>Tract demographics</div>
      <>
        <J40AccordionItem id={'race'} title={`Racial Ethnographic`}>
          {demographicItemGen(demographicsData.racial)}
        </J40AccordionItem>
        <J40AccordionItem id={'age'} title={`Age`}>
          {demographicItemGen(demographicsData.age)}
        </J40AccordionItem>
      </>
    </div>
  );
};

export default TractDemographics;
