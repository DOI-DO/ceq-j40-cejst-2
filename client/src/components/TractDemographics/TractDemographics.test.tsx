import React from 'react';
import { render } from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import TractDemographics from './TractDemographics';

describe('rendering of TractDemographics Component', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <TractDemographics />
        </LocalizedComponent>,
    );
    it('checks if component renders', () => {
      expect(asFragment()).toMatchSnapshot();
    });
});
