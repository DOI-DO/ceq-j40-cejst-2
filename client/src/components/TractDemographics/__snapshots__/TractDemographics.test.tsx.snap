// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of TractDemographics Component checks if component renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      Tract demographics
    </div>
    <h6>
      Racial Ethnographic
      <button
        aria-controls="race-panel"
        aria-expanded="false"
        id="race-header"
        tabindex="0"
      >
        show
      </button>
      <img
        alt="collapse icon"
        src="test-file-stub"
      />
    </h6>
    <section
      aria-labelledby="race-header"
      hidden=""
      id="race-panel"
    >
      <div>
        <span>
          White
        </span>
        <span>
          61.4%
        </span>
      </div>
      <div>
        <span>
          Black or African Americon
        </span>
        <span>
          10.3%
        </span>
      </div>
      <div>
        <span>
          American Indian and Alaska Native
        </span>
        <span>
          10.3%
        </span>
      </div>
      <div>
        <span>
          American Indian and Alaska Native
        </span>
        <span>
          0.3%
        </span>
      </div>
      <div>
        <span>
          Asian
        </span>
        <span>
          7.2%
        </span>
      </div>
      <div>
        <span>
          Native Hawiian or Pacific Islander
        </span>
        <span>
          4.2%
        </span>
      </div>
      <div>
        <span>
          Other
        </span>
        <span>
          2.5%
        </span>
      </div>
      <div>
        <span>
          Two or more races
        </span>
        <span>
          6.7%
        </span>
      </div>
      <div>
        <span>
          Hispanic or Latino
        </span>
        <span>
          10.4%
        </span>
      </div>
    </section>
    <h6>
      Age
      <button
        aria-controls="age-panel"
        aria-expanded="false"
        id="age-header"
        tabindex="0"
      >
        show
      </button>
      <img
        alt="collapse icon"
        src="test-file-stub"
      />
    </h6>
    <section
      aria-labelledby="age-header"
      hidden=""
      id="age-panel"
    >
      <div>
        <span>
          Children under 10
        </span>
        <span>
          1.4%
        </span>
      </div>
      <div>
        <span>
          Ages 10 - 64
        </span>
        <span>
          80.3%
        </span>
      </div>
      <div>
        <span>
          Elderly over 65
        </span>
        <span>
          7.2%
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
